lib/rubocop/cop/rails/private_transaction_option.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Checks whether `ActiveRecord::Base.transaction(joinable: _)` is used.
+      #
+      # The `joinable` option is a private API and is not intended to be called
+      # from outside Active Record core.
+      # https://github.com/rails/rails/issues/39912#issuecomment-665483779
+      # https://github.com/rails/rails/issues/46182#issuecomment-1265966330
+      #
+      # Passing `joinable: false` may cause unexpected behavior such as the
+      # `after_commit` callback not firing at the appropriate time.
+      #
+      # @safety
+      #   This Cop is unsafe because it cannot accurately identify
+      #   the `ActiveRecord::Base.transaction` method call.
```

Comment:
Well, I bet `transaction(joinable: true/false)` would be a Rails transaction call with very high certainty.

I'd remove this clause and made the cop safe.

I would accept a single case of such a false positive from https://github.com/jeromedalbert/real-world-ruby-apps as a counter-argument.

Reply:
Hmm, this is up for debate. Personally, I believe it should be marked as unsafe as long as there is theoretically a possibility of false positives.

I'd be interested to hear from other maintainers.

Reply:
It seems that the issue is not a false positive in the cop, but rather the change in behavior caused by the autocorrect. Therefore, shouldn't it be `SafeAutoCorrect: false` instead of `Safe: false`?